Make schema warnings available through navbar

bjoerge opened the pull request:
### Description
Currently we do not show any schema warnings in v3 studios *unless* there's also schema errors. This adds a button to easily view schema errors to the global navbar. This is only visible in dev mode (so you won't be able to see it in action on the preview deployment for this PR, but it it is enabled in this build: https://test-studio-itunv5byl.sanity.build/)
![image](https://user-images.githubusercontent.com/876086/184146235-0b189090-01cd-4a39-96d2-f307dadf8439.png)


### Notes for release
- Schema warnings will now be accessible via the global nav bar in development mode

rexxars commented on packages/sanity/src/studio/screens/schemaErrors/SchemaProblemGroups.tsx:
```diff
@@ -0,0 +1,133 @@
+export function SchemaProblemGroups(props: {problemGroups: SchemaValidationProblemGroup[]}) {
+  const {problemGroups} = props
+
+  const items = useMemo(() => {
+    const ret = []
+
+    for (const problemGroup of problemGroups) {
+      for (const problem of problemGroup.problems) {
+        ret.push({group: problemGroup, problem})
+      }
+    }
+
+    return ret
+  }, [problemGroups])
```
Might be easier to use `React.memo` on the component here, since it is the only prop received?

bjoerge replied:
This was already there from earlier (this file has been slightly modified + renamed and git doesn't seem to have picked it up).

rexxars commented on packages/sanity/src/studio/components/navbar/schemaWarnings/SchemaWarningsButton.tsx:
```diff
@@ -0,0 +1,101 @@
+      group.problems.some((problem) => problem.severity === 'warning')
+    ) || []
+
+  const popoverProps = useMemo(
+    () => ({
+      constrainSize: true,
+      portal: true,
+      scheme: scheme,
+    }),
+    [scheme]
+  )
```
I can't see this used anywhere?

bjoerge replied:
Copy paste err, thanks for spotting!

rexxars commented on packages/sanity/src/studio/components/navbar/schemaWarnings/SchemaWarningsButton.tsx:
```diff
@@ -0,0 +1,101 @@
+      ),
+      onClickOutside: handleClose,
+      onClose: handleClose,
+      scheme: 'light',
```
Always light? If intentional, can we put a comment here to explain why?

bjoerge replied:
Copy paste from the ChangelogDialog - I've asked @hermanwikner if he knows why we need it there - will remove & update with a comment.

bjoerge replied:
Ah - I realised now why it's needed: the navbar is dark schemed by default, so the dialog will inherit dark mode from it. That's an interesting issue - I guess what we'd want to express here is more like `scheme: "opposite"` or `scheme: "reset"` or something… 🤔

rexxars reviewed (changes requested):
This is a fantastic idea! Couple of thoughts:

1. This message is helpful, but hard to spot at the bottom:
_Note: this message is only displayed during development_
Perhaps we can move it to the top to be gain more visibility?

2. I would love for the warnings to be sorted somehow - I guess they might implicitly be sorted by schema type? If so, I suppose that's alright! If not, sorting by schema type or type of error would be amazing.
3. I could see us having _config warnings_, so it would be nice if the dialog reflected this instead of explicitly saying "schema errors".
4. The "schema warning" header on each warning is a bit repetitive - perhaps we could instead show a "Schema warnings" header above the list?

hermanwikner commented on packages/sanity/src/studio/components/navbar/schemaWarnings/SchemaWarningsButton.tsx:
```diff
@@ -0,0 +1,101 @@
+          {...dialogProps}
+          id={dialogId}
+        >
+          <Stack space={4} padding={3}>
```
This `Stack` is unnecessary since the `SchemaProblemsGroup` has its once `Stack`. 

I suggest changing it to a `Box` and bump to `padding` to `4`!

```js
<Box padding={4}>
  <SchemaProblemGroups problemGroups={groupsWithWarnings} />
</Box>
```

hermanwikner commented:
The logo is not centered in mobile mode. I suggest, as with the changelog, to not show these warnings in mobile mode. This could be fixed by updating the `shouldRender` condition to `mediaIndex > 1 && isDev` in `Navbar` :)

<img width="868" alt="Screenshot 2022-08-12 at 10 41 21" src="https://user-images.githubusercontent.com/15094168/184320916-d7ae490a-9b4e-4373-bde2-72fdcf99e972.png">

bjoerge commented:
Great feedback, thanks!

> 1. Perhaps we can move it to the top to be gain more visibility?

Like so? 
<img width="1045" alt="image" src="https://user-images.githubusercontent.com/876086/184319029-6384e8f4-9d3b-4712-b200-ec5a8837529a.png">

> 2. I would love for the warnings to be sorted somehow

I think they are currently appearing in the same order they are added to the schema. Each schema type are grouped together. I can sort them alphabetically by schema type name if you think that's better. The underlying data structure we use here groups all issues from the same "path" together, but at some point a decision was made to list them out flat/ungrouped in this view. Let's improve on this in a future iteration?

> 3. would be nice if the dialog reflected this instead of explicitly saying "schema errors".

I've changed the name of this thing to be "Configuration issues" now. Makes a lot of sense - great suggestion!

> 4. The "schema warning" header on each warning is a bit repetitive - perhaps we could instead show a "Schema warnings" header above the list?

I've changed the title to display the name and type of the schema type with the issue, I think helps, although it still repeats for multiple warnings for the same schema type, but given that we decided at some point do show these ungrouped I'm hesitant to do too much work in this area.

<img width="688" alt="image" src="https://user-images.githubusercontent.com/876086/184321923-81fe2af6-be20-48cc-9870-5ce72c2ff5a9.png">

bjoerge commented:
> The logo is not centered in mobile mode.

Great suggestion! Fixed.

bjoerge commented:
@rexxars, @hermanwikner Here's a new deployment with your feedback addressed: https://test-studio-no0kiw57n.sanity.build/

hermanwikner reviewed (approved):
💯

rexxars reviewed (approved):
👍 💯
